components/match2/commons/match.lua

```diff
@@ -313,7 +313,7 @@ function Match._prepareRecordsForStore(records)
 	for opponentIndex, opponentRecord in ipairs(records.opponentRecords) do
 		Match.clampFields(opponentRecord, Match.opponentFields)
 		for _, playerRecord in ipairs(records.playerRecords[opponentIndex]) do
-			Match.clampFields(playerRecord, Match.playerFields)
+			Match._preparePlayerRecordForStore(playerRecord)
 		end
```

Comment:
imo use `Array.forEach(records.playerRecords[opponentIndex], Match._preparePlayerRecordForStore)` instead of the entire ipairs/for loop

Reply:
I'm keeping existing style for now, there are other loops around doing the same